web-registry/src/components/organisms/WalletButton/WalletButton.tsx

```diff
@@ -29,6 +33,15 @@ const WalletButton: React.FC = () => {
     useState<WalletModalState>('wallet-select');
   const isConnected = loaded ? !!wallet?.address : null;
 
+  // Populate cache with user balance once connected
+  useQuery(
+    getAllBalancesQuery({
```

Comment:
you could just use the `Balance` query which would avoid using `find` afterwards

also, ideally we could also refetch the balance as soon as we try to initiate a new tx
indeed right now, if I top up the regen account balance and then retry, I still get the warning, we need a page refresh to clear the cache and make it disappear

Reply:
I swapped the query to `Balance` and refetch the cache before checking the balance for the fees.